x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ReplaceDataStreamBackingIndexStep.java

```diff
@@ -75,7 +75,7 @@ public ClusterState performAction(Index index, ClusterState clusterState) {
         }
 
         assert dataStream.getWriteIndex() != null : dataStream.getName() + " has no write index";
-        if (dataStream.getWriteIndex().getIndex().equals(originalIndex)) {
+        if (dataStream.getWriteIndex().getIndex().getName().equals(originalIndex)) {
```

Comment:
on second thought, to be consistent, I believe we should check indices equality (not just the name)
```suggestion
        if (dataStream.getWriteIndex().getIndex().equals(index)) {
```
We already do this in [another step](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CheckNotDataStreamWriteIndexStep.java#L64)